src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_StackAllocArrayCreationAndInitializer.cs

```diff
@@ -0,0 +1,705 @@
+      ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 1, IsInvalid) (Syntax: '1')
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // file.cs(6,27): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
+                //         var a = /*<bind>*/stackalloc int[1]/*</bind>*/;
+                Diagnostic(ErrorCode.ERR_UnsafeNeeded, "stackalloc int[1]").WithLocation(6, 27)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<StackAllocArrayCreationExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact]
+        public void SimpleStackAllocArrayCreation_UserDefinedType()
+        {
+            string source = @"
+struct M { }
+
+class C
+{
+    public void F()
+    {
+        var a = /*<bind>*/stackalloc M[1]/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+IOperation:  (OperationKind.None, Type: null, IsInvalid) (Syntax: 'stackalloc M[1]')
+  Children(1):
+      ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 1, IsInvalid) (Syntax: '1')
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // file.cs(8,27): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
+                //         var a = /*<bind>*/stackalloc M[1]/*</bind>*/;
+                Diagnostic(ErrorCode.ERR_UnsafeNeeded, "stackalloc M[1]").WithLocation(8, 27)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<StackAllocArrayCreationExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact]
+        public void SimpleStackAllocArrayCreation_ConstantDimension()
+        {
+            string source = @"
+struct M { }
+
+class C
+{
+    public void F()
+    {
+        const int dimension = 1;
+        var a = /*<bind>*/stackalloc M[dimension]/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+IOperation:  (OperationKind.None, Type: null, IsInvalid) (Syntax: 'stackalloc M[dimension]')
+  Children(1):
+      ILocalReferenceOperation: dimension (OperationKind.LocalReference, Type: System.Int32, Constant: 1, IsInvalid) (Syntax: 'dimension')
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // file.cs(9,27): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
+                //         var a = /*<bind>*/stackalloc M[dimension]/*</bind>*/;
+                Diagnostic(ErrorCode.ERR_UnsafeNeeded, "stackalloc M[dimension]").WithLocation(9, 27)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<StackAllocArrayCreationExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact]
+        public void SimpleStackAllocArrayCreation_NonConstantDimension()
+        {
+            string source = @"
+struct M { }
```

Comment:
Nit: in the future, consider naming `S` (more conventional)

Reply:
(inherited from array creation tests)